src/lib.rs

```diff
@@ -632,3 +660,20 @@ impl Buffer {
         &self.snappy[..snappy_len]
     }
 }
+
+/// handler with weak reference to layer quit notify
```

Comment:
Should these doc comments start with `C`apital and end with `.`? will match existing doc comments on this file AFAICT

Reply:
English is not my main language, this is my best, feel free to change after merge if code is ok.